### ISSUE TITLE
Problem comparing Jsons

### ISSUE THREAD

**Issue by AndreaCimminoArriaga** [farshidtz/wot-discovery-testing#6]
I got the following FAIL, however the unique difference between the expected and actually retrieved TD is the order of the elements in the array forms
`````
--- FAIL: TestPatch/replace_array/result (0.02s)
            registration_test.go:837: Expected:
                {
                	"@context": "https://www.w3.org/2019/wot/td/v1",
                	"id": "urn:uuid:b537f14f-2363-4eb5-b3d4-d25b9a3a2df4",
                	"properties": {
                		"status": {
                			"forms": [
                				{
                					"href": "https://mylamp.example.com/status"
                				},
                				{
                					"href": "coaps://mylamp.example.com/status"
                				}
                			]
                		}
                	},
                	"security": [
                		"nosec_sc"
                	],
                	"securityDefinitions": {
                		"nosec_sc": {
                			"scheme": "nosec"
                		}
                	},
                	"title": "example thing"
                }
                 Retrieved:
                {
                	"@context": "https://www.w3.org/2019/wot/td/v1",
                	"id": "urn:uuid:b537f14f-2363-4eb5-b3d4-d25b9a3a2df4",
                	"properties": {
                		"status": {
                			"forms": [
                				{
                					"href": "coaps://mylamp.example.com/status"
                				},
                				{
                					"href": "https://mylamp.example.com/status"
                				}
                			]
                		}
                	},
                	"security": [
                		"nosec_sc"
                	],
                	"securityDefinitions": {
                		"nosec_sc": {
                			"scheme": "nosec"
                		}
                	},
                	"title": "example thing"
                }
`````

**Comment by ethieblin** [comment]
We have the same issue

**Comment by farshidtz** [comment]
Resolved by #29
